Codegen bug with exportc and case objects

Test case
```nim
type
  Helper* {.exportc: "PublicHelper".} = object
    case isKind: bool
      of true:
        formatted: string
      of false:
        parsed1: string
        parsed2: string

proc test(c: Helper): string =
  c.formatted

echo test(Helper(isKind: true, formatted: "ok"))
```

On latest Nim generated C fails to compile with error:
```
Error: execution of an external compiler program 'gcc.exe -c  -w -mno-ms-bitfields   -IC:\Nim\lib -IC:\Nim -o "C:\t1_d\@mt1.nim.c.o" "C:\t1_d\@mt1.nim.c"' failed with exit code: 1

C:\t1_d\@mt1.nim.c: In function 'test__cf7ypsLB70bFkcUMOU1MAg':
C:\t1_d\@mt1.nim.c:120:23: error: 'PublicHelper' {aka 'struct PublicHelper'} has no member named 'formatted'
  result = copyString(c.formatted);
```
Compiles fine without `exportc` pragma

I have a fix

--- a/compiler/semtypes.nim
+++ b/compiler/semtypes.nim
@@ -698,7 +698,7 @@ proc semRecordCase(c: PContext, n: PNode, check: var IntSet, pos: var int,
   father.add a
 
 proc semRecordNodeAux(c: PContext, n: PNode, check: var IntSet, pos: var int,
-                      father: PNode, rectype: PType, hasCaseFields = false) =
+                      father: PNode, rectype: PType, hasCaseFields: bool) =
   if n == nil: return
   case n.kind
   of nkRecWhen:
@@ -727,12 +727,12 @@ proc semRecordNodeAux(c: PContext, n: PNode, check: var IntSet, pos: var int,
         assign(newCheck, check)
         var newPos = pos
         var newf = newNodeI(nkRecList, n.info)
-        semRecordNodeAux(c, it[idx], newCheck, newPos, newf, rectype)
+        semRecordNodeAux(c, it[idx], newCheck, newPos, newf, rectype, hasCaseFields)
         it[idx] = if newf.len == 1: newf[0] else: newf
     if c.inGenericContext > 0:
       father.add n
     elif branch != nil:
-      semRecordNodeAux(c, branch, check, pos, father, rectype)
+      semRecordNodeAux(c, branch, check, pos, father, rectype, hasCaseFields)
   of nkRecCase:
     semRecordCase(c, n, check, pos, father, rectype)
   of nkNilLit:
@@ -741,7 +741,7 @@ proc semRecordNodeAux(c: PContext, n: PNode, check: var IntSet, pos: var int,
     # attempt to keep the nesting at a sane level:
     var a = if father.kind == nkRecList: father else: copyNode(n)
     for i in 0..<n.len:
-      semRecordNodeAux(c, n[i], check, pos, a, rectype)
+      semRecordNodeAux(c, n[i], check, pos, a, rectype, hasCaseFields)
     if a != father: father.add a
   of nkIdentDefs:
     checkMinSonsLen(n, 3, c.config)

--- a/tests/ccgbugs/tcgbug.nim
+++ b/tests/ccgbugs/tcgbug.nim
@@ -40,7 +40,8 @@ type
 var k = PFuture[void]()
 
 
-##bug #9297
+##bug #9297 and #13281
+
 import strutils
 
 type

--- a/tests/ccgbugs/tcgbug.nim
+++ b/tests/ccgbugs/tcgbug.nim
@@ -2,6 +2,7 @@ discard """
 output: '''
 success
 M1 M2
+ok
 '''
 """
 
@@ -58,6 +59,14 @@ type
       of M2: b:float
       of M3: c:cstring
 
+  Helper* {.exportc: "PublicHelper".} = object
+    case isKind: bool
+      of true:
+        formatted: string
+      of false:
+        parsed1: string
+        parsed2: string
+
 proc newMyObject(kind: MyKind, val: string): MyObject =
   result = MyObject(kind: kind)
 
@@ -75,3 +84,9 @@ proc newMyObjectRef(kind: MyKind, val: string): MyObjectRef =
 
 
 echo newMyObject(M1, "2").kind, " ", newMyObjectRef(M2, "3").kind
+
+
+proc test(c: Helper): string =
+  c.formatted
+
+echo test(Helper(isKind: true, formatted: "ok"))

